include/ceed.h

```diff
@@ -1 +1,5 @@
+#ifdef CEED_RUNNING_JIT_PASS
+#include "ceed/types.h"
+#else
 #include "ceed/ceed.h"
+#endif
```

Comment:
This should be transparent to the user

Reply:
Yeah, though the IDE (`clangd` or other) will assume you get the whole `ceed/ceed.h` and thus not notify if you use functions that should not be used in qfunctions. So this is fine to preserve compatibility, but we should also move toward only including `ceed/types.h` so that qfunction sources are checked in a similar context to how they will be during JIT (where it's harder to work with error messages).

Reply:
agreed (latest commit)